src/python/dxpy/utils/app_unbuilder.py

```diff
@@ -46,6 +47,21 @@ def _recursive_cleanup(foo):
             if val == "" or val == [] or val == {}:
                 del foo[key]
 
+def flatten_patterns(metadata):
+    """
+    Flattens string representations of 'inputSpec.patterns' and
+    'outputSpec.patterns' arrays using regular expressions.
+
+    :param metadata: Executable metadata
+    :type metadata: str
+    """
+
+    result = re.sub("\"patterns\": \\[\r?\n\\s*", "\"patterns\": [", metadata, flags=re.MULTILINE)
```

Comment:
You can use single quotes to quote the string here, so as to avoid having to escape all the interior quotes.